### PR TITLE
Add iOS release instructions page

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

**Comment by cbracken** [comment]
I'll spend some time cleaning up and improving the wording tonight and tomorrow, but sending out a rough first draft that hits all the critical steps.

**Comment by eseidelGoogle** [comment]
This is great.  Thank you.  Plenty of information to set folks on the right track.

File: ios-release-build.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+layout: page
+title: Preparing an iOS App for Release
+sidebar: home_sidebar
+permalink: /iOS-release/
```
Comment (mit-mit) [inline ios-release-build.md]: I believe so far we have used all lower-case for our permalinks

File: ios-release-build.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+navigator. Then, in the main view sidebar, select the `Runner` target and open
+the `General` tab.
+
+In the `Signing` section, check the `Automatically manage signing` checkbox,
+then select your registered development team from the `Team` pop-up menu.
+
+For a detailed overview of app signing, see [Maintaining Your Signing
+Identities and Certificates][appsigning] in Apple's App Distribution Guide.
+
+## Creating a build archive
```
Comment (mit-mit) [inline ios-release-build.md]: Isn't this a substep of the previous `## Register your app on iTunes Connect` section?

File: ios-release-build.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  * From Xcode: Invoke `File > Open` and select the `ios` folder located inside
+    your main app directory.
+
+  * From the terminal: `cd <app dir>; open ios/Runner.xcworkspace`.
+
+* TOC Placeholder
+{:toc}
+
+
+## Register your app on iTunes Connect
```
Comment (mit-mit) [inline ios-release-build.md]: Maybe move this section down to after `## Signing the app` to make it flow more chronologically?

File: ios-release-build.md (hunk 1 above)
Comment (cbracken) [inline ios-release-build.md]: Done.

File: ios-release-build.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+sidebar: home_sidebar
+permalink: /iOS-release/
+---
+
+During the typical development cycle, you'll test an app using `flutter run` at
+the command line, the Run and Debug toolbar buttons in IntelliJ, or both. By
+default, Flutter builds a *debug* version of your app.
+
+When you're ready to prepare a *release* version, for example to [publish to
+the App Store][appstore], follow the steps on this page. To release an app to
```
Comment (cbracken) [inline ios-release-build.md]: Good point. Broke it out into a small section to make it stand out as a separate point.

File: ios-release-build.md (hunk 3 above)
Comment (cbracken) [inline ios-release-build.md]: Creating a bundle ID needs to happen before app signing. Registering the application record is pretty closely related to creating the bundle ID, and both happen on the same site, so tempted to keep them together as part of the same larger step.

File: ios-release-build.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+* TOC Placeholder
+{:toc}
+
+
+## Register your app on iTunes Connect
+
+Before uploading a build to the App Store, you'll need to register your app on
+[iTunes Connect][itunesconnect]. iTunes Connect is where you'll manage your
+application's metadata, pricing and releases to the App Store.
```
Comment (cbracken) [inline ios-release-build.md]: /shamecube. I'm usually the guy shouting at others with enthusiastic/desperate endorsements of the Oxford comma. After last week's [NY Times](https://www.nytimes.com/2017/03/16/us/oxford-comma-lawsuit.html?_r=0) article, we've got more ammo on our side! Thanks for catching this.

Did a quick scan. It looks like there aren't any others... *phew*.

File: ios-release-build.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+1. Enter an app name and select `Explicit App ID` and enter an ID.
+
+1. Select any services, then click `Continue`.
+
+1. On the next page, confirm the details and click `Register` to register your
+   Bundle ID.
+
+### Create an iTunes Connect application record
+
+The next step is to register your app on iTunes Connect. Follow these steps:
```
Comment (cbracken) [inline ios-release-build.md]: Covered above (ll. 36-37) in the `##` heading for the overall iTunes Connect step, under which the bundle ID and this subsection are nested.

File: ios-release-build.md (hunk 2 above)
Comment (cbracken) [inline ios-release-build.md]: Those steps happen in a browser on Apple's iTunes Connect site, this step is a build step that happens in Xcode.

File: ios-release-build.md (hunk 6 above)
Comment (cbracken) [inline ios-release-build.md]: I've done some cleanup on that section, made it a bit more visible by pushing it to the front of the section, and added a bit of detail. Let me know if that reads a bit better now.

File: ios-release-build.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+Identities and Certificates][appsigning] in Apple's App Distribution Guide.
+
+## Creating a build archive
+
+This section describes how to create a build archive and upload your build to
+iTunes Connect.
+
+Using the command line:
+
+1. `cd <app dir>` (replace `<app dir>` with your application's directory).
```
Comment (xster) [inline ios-release-build.md]: Do we need this in the various places? I think we make this assumption in other pages already.

**Comment by cbracken** [comment]
Landed a bunch of cleanup and added links to useful sections of the App Distribution Guide.

Done with the overall structure. The text should be a fair bit cleaner now.

@LarkAscending @kwalrath @Sfshaza I suspect this is ready for your expert review. It'd also be useful to know where you think this would benefit most from screenshots.

File: ios-release-build.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+the App Store. For a detailed overview, see Apple's [App Distribution
+Guide][distributionguide].
+
+*Note*: Many of the steps below must be carried out in Xcode. To open your app
+in Xcode:
+
+  * From Xcode: Invoke `File > Open` and select the `ios` folder located inside
+    your main app directory.
+
+  * From the terminal: `cd <app dir>; open ios/Runner.xcworkspace`.
```
Comment (cbracken) [inline ios-release-build.md]: Good catch. I've replaced this with instructions to open the workspace from Xcode, which is likely the simpler thing to do in any case.

File: ios-release-build.md (hunk 7 above)
Comment (cbracken) [inline ios-release-build.md]: Deleted!

File: ios-release-build.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
+---
+layout: page
+title: Preparing an iOS App for Release
+sidebar: home_sidebar
+permalink: /ios-release/
+---
+
+During development, you'll test an app using `flutter run` at the command line,
+using the Run and Debug toolbar buttons in IntelliJ, or both. By default,
+Flutter builds a *debug* version of your app.
```
Comment (cbracken) [inline ios-release-build.md]: I wonder if this paragraph adds anything? Maybe we should just kill it and go directly to the point?

Comment (xster) [inline ios-release-build.md]: I think it's still useful but maybe don't lead with it. Move it next to the flutter build line in the middle?

File: ios-release-build.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+### Create an application record on iTunes Connect
+
+Next, you'll register your app on iTunes Connect:
+
+1. Open [iTunes Connect][itunesconnect_login] in your browser.
+1. On the iTunes Connect landing page, click `My Apps`.
+1. Click '+' in the top-left corner of the `My Apps` page, then select `New
+   App`.
+1. Fill in your app details on the form that appears. In the `Platforms`
+   section, ensure that iOS is checked. Leave tvOS unchecked. Click `Create`.
```
Comment (xster) [inline ios-release-build.md]: Mini explanation that tv is not yet supported?

File: ios-release-build.md [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+   Information` from the sidebar.
+1. In the `General Information` section, select the Bundle ID you registered in
+   the preceding step.
+
+For a detailed overview, see [Creating an iTunes Connect Record for an
+App][itunesconnect_guide_register].
+
+## Review Xcode project settings
+
+Before you create a build, it's important to ensure that the application
```
Comment (xster) [inline ios-release-build.md]: Slight verbosity

File: ios-release-build.md (hunk 9 above)
Comment (cbracken) [inline ios-release-build.md]: Good call - done!

File: ios-release-build.md (hunk 10 above)
Comment (cbracken) [inline ios-release-build.md]: Done.

File: ios-release-build.md (hunk 11 above)
Comment (cbracken) [inline ios-release-build.md]: Good point, this sentence adds no real value over the following paragraph. Deleted.

File: ios-release-build.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+In the `Identity` section:
+
+  * `Display Name`: the name of the app to be displayed on the home screen and
+    elsewhere.
+  * `Bundle Identifier`: the App ID you registered on iTunes Connect.
+
+In the `Signing` section:
+
+  * `Automatically manage signing`: whether Xcode should automatically manage
+    app signing and provisioning. This is set `true` by default.
```
Comment (nhcohen) [inline ios-release-build.md]: Any advice on when this should be set to true, and when to false?

File: ios-release-build.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+1. In the sidebar of the Xcode Organizer window, select your iOS app, then
+   select the build archive you just produced.
+1. Click the `Validate...` button. If any issues are reported, address them and
+   produce another build -- you can reuse the same build ID until you upload an
+   archive.
+1. Once you've successfully validated an archive, click `Upload to App
+   Store...`. You can follow the status of your build in the `Activities` tab
+   of your app's details page on [iTunes Connect][itunesconnect_login].
+
+You should receive an email within a half-hour notifying you when your build
```
Comment (nhcohen) [inline ios-release-build.md]: "when" => "that"

File: ios-release-build.md [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+Release` section.
+
+For more details, see [Submitting Your App to the
+Store][distributionguide_submit].
+
+## Troubleshooting
+
+The [App Distribution Guide][distributionguide] provides a detailed overview of
+the process of releasing an app to the App Store. It includes a
+[Troubleshooting][distributionguide_troubleshooting] guide that includes
```
Comment (nhcohen) [inline ios-release-build.md]: "[Troubleshooting][distributionguide_troubleshooting] guide" => 
"[Troubleshooting guide][distributionguide_troubleshooting] "

File: ios-release-build.md (hunk 5 above)
Comment (mit-mit) [inline ios-release-build.md]: :-)

File: ios-release-build.md (hunk 6 above)
Comment (mit-mit) [inline ios-release-build.md]: New version works great!

File: ios-release-build.md (hunk 12 above)
Comment (cbracken) [inline ios-release-build.md]: Done.

File: ios-release-build.md (hunk 13 above)
Comment (cbracken) [inline ios-release-build.md]: Done.

File: ios-release-build.md (hunk 14 above)
Comment (cbracken) [inline ios-release-build.md]: Done.

**Comment by mit-mit** [comment]
@cbracken can you give me a ping when you merge this? Then I'll do another PR to add links to this and the Android one.

**Comment by cbracken** [comment]
Merged. :)

**Pull request merged**